receiver/prometheusreceiver/metrics_receiver_test.go

```diff
@@ -79,44 +105,205 @@ func TestNew(t *testing.T) {
 	}
 }
 
+// Test data for EndToEnd test
+
+var target1Page1 = `
```

Comment:
@fivesheep - yes, the endtoend test does help clarify things. A couple of comments:
1. Adding a some comments on the target pages about what's being tested would be helpful.
2. It looks like resets aren't tested - adding that to a test page would also be worthwhile
3. It also looks like empty histogragms/summaries aren't tested here either - I think they would be worthwhile to have too.

Reply:
sure will do.  
I just checked the reset cases, it seems like when it resets, the adjuster will discard the value (use it as initial) when a reset is detected.  Is it an expected behavior? I thought it would use the timestamp from previous scrape as starttime, and value as it is (starting from 0). it's not right or wrong here, just a decision of best-effort.  if this is what we want, I can keep the test case this way.

Reply:
Yes, a reset acts like an initial point since we don't know when the reset actually occurred. Using the timestamp from the previous scrape does give a much more constrained lower bound than the initial point and may be worth considering - I need to think through the implications.

@rghetia - thoughts?

Reply:
It is better to treat reset as initial point. Consider this
0. At time T0 a job reported data
1. At time T1 a job stops reporting for whatever reason.
2. At time T2 the job resets
3. At time T3 the job resumes reporting.

If time lag between T0 and T3 is within scraping interval then it is okay to use T0 as a starttime. Otherwise it will provide incorrect data. I would always prefer no-data vs incorrect data.